sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/FileSystem.java

```diff
@@ -101,7 +101,9 @@
    * @throws IllegalArgumentException is File is null or not a directory.
    * 
    * @since 4.5
+   * @deprecated since 6.6 Ability to report issues or measures on directories will soon be dropped. Report issues on project if needed. 
    */
+  @Deprecated
   @CheckForNull
   InputDir inputDir(File dir);
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)